packages/gatsby/src/schema/infer/add-inferred-fields.js

```diff
@@ -293,18 +231,26 @@ const getFieldConfigFromFieldNameConvention = ({
   // (ii) hinders reusing types.
   if (linkedTypes.length > 1) {
     const typeName = linkedTypes.sort().join(``) + `Union`
-    type = schemaComposer.getOrCreateUTC(typeName, utc => {
-      const types = linkedTypes.map(typeName =>
-        schemaComposer.getOrCreateOTC(typeName)
-      )
-      utc.setTypes(types)
-      utc.setResolveType(node => node.internal.type)
+    type = UnionTypeComposer.createTemp({
```

Comment:
Do we still have to use a temporary composer now that we choose from `possibleFields` earlier?

Reply:
Everything in add-inferred-fields is a temp composer, it's only added in merge.

Reply:
ah cool, just asking in general if we still need to work with temporary composers (instead of directly mutating)? the original motivation for temp composers was because we inferred for all fields even if they had conflicting field names, and only then decided which one should have priority. now that we decide before inferring do we still have to do it? could remove a bit of complexity, e.g. [here](https://github.com/gatsbyjs/gatsby/blob/774ada430f6dc67378facc6e2493976740f84f4f/packages/gatsby/src/schema/infer/add-inferred-fields.js#L297-L317). if we still need to do it, maybe a short comment in `addInferredType`?